packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -556,7 +563,15 @@ export class GarbageCollector implements IGarbageCollector {
              * but make it one day to be safe.
              */
             if (snapshotCacheExpiryMs !== undefined) {
-                this.sweepTimeoutMs = this.sessionExpiryTimeoutMs + snapshotCacheExpiryMs + oneDayMs;
+                const overrideSweepTimeoutMs =
+                    this.mc.config.getNumber("Fluid.GarbageCollection.TestOverride.SweepTimeoutMs");
+
+                this.sweepTimeoutMs =
+                    overrideSweepTimeoutMs
+                    ?? this.sessionExpiryTimeoutMs + snapshotCacheExpiryMs + oneDayMs;
+
+                assert(this.sweepTimeoutMs > this.sessionExpiryTimeoutMs + snapshotCacheExpiryMs,
```

Comment:
It seems overriding sweep timeout is rather complicated. I would suggest writing an end-to-end test that would use these configs to validate that sweep works correctly.
IMO, it will be difficult to write tests with session expiry and sweep timeout both enabled since the summarizer would close before it can run sweep. I would imagine most tests to be simple and validate sweep logic without session expiry.

Reply:
I'm trying out Tyler's suggestion which is to maintain the calculation as-is and instead support shrinking buffer.

> I would imagine most tests to be simple and validate sweep logic without session expiry.

I want to avoid opening up holes in what the code supports for easier testing.  I am expecting to write an e2e test that:

1. Unreferences an object
2. Waits SessionExpiry time and sees that container close
3. Opens a new container
4. After SweepTimeout has elapsed ensure it's marked as SweepReady.

Reply:
That sounds fine. I would however recommend adding an end-to-end test with this PR to get a sense of how this will work and also to validate that this works as expected with session timeout. I believe unit test is not enough to validate it.
While I agree that we should not be opening up holes to support testing, sometimes we can do things a little differently based on the results of tests :-).